### DISCUSSION TITLE
Adding of HTML fragment with CSS and JS

### DISCUSSION THREAD

**Discussion by denishsharma** [retypeapp/retype#265] [Ideas]
It would be nice if we could render custom HTML fragments with CSS and JS. As a UI designer, documenting the components and its state is crucial. Another option is to integrate the StoryBook fragment.

For example, if I want to interact with a button component then I will simply add HTML file which I already coded.
`[!html src="../static/component/button.html"]`

Or embedding the story book
`[!storybook src="https://5ccbc373887ca40020446347-bysekhynzd.chromatic.com/?path=/story/shadowboxcta--default&full=1&shortcuts=false&singleStory=true"]`

**Comment by fabriciomurta** [comment]
Sounds like at least part of this will be possible with upcoming templating, variables and partials support. There's some discussion around it in #213, you may want to check it out.
